unified_planning/model/ma_problem.py

```diff
@@ -0,0 +1,250 @@
+        self._initial_value[fluent_exp] = value_exp
+
+
+    def initial_values(self) -> Dict['unified_planning.model.fnode.FNode', 'unified_planning.model.fnode.FNode']:
+        '''Gets the initial value of the fluents.'''
+        res = self._initial_value
+        return res
+
+    def add_fluent(self, fluent: 'unified_planning.model.fluent.Fluent', *,
+                   default_initial_value: Union['unified_planning.model.fnode.FNode', 'unified_planning.model.object.Object', bool,
+                                                int, float, Fraction] = None):
+        '''Adds the given fluent.'''
+        if self.has_name(fluent.name()):
+            raise UPProblemDefinitionError('Name ' + fluent.name() + ' already defined!')
+        self._fluents.append(fluent)
+        if not default_initial_value is None:
+            v_exp, = self._env.expression_manager.auto_promote(default_initial_value)
+            self._fluents_defaults[fluent] = v_exp
+        '''if fluent.type().is_user_type() and fluent.type() not in self._user_types:
+            self._user_types.append(fluent.type()) # type: ignore
+        for type in fluent.signature():
+            if type.is_user_type() and type not in self._user_types:
+                self._user_types.append(type)'''
+
+
+    def add_objects(self, objs: List['unified_planning.model.object.Object']):
+        '''Adds the given fluents.'''
+        #controllare##############
+        for obj in objs:
+
+            if self.has_name(obj.name()):
+                raise UPProblemDefinitionError('Name ' + obj.name() + ' already defined!')
+            self.add_object(obj)
+            if obj.type().is_user_type() and obj.type() not in self._user_types:
+                self._user_types.append(obj.type())
+
+    def add_user_types(self, user_types):
+        for user_type in user_types:
+            if user_type.is_user_type() and user_type not in self._user_types:
+                self._user_types.append(user_type)
+
+    '''def add_objects(self, objs: List['unified_planning.model.object.Object']):
+        for ag in self.agents_list:
+            for obj in objs:
+                obj = copy.deepcopy(obj)
+                setattr(obj, '_name', str(getattr(obj, '_name')) + "_" + str(self.get_agents().index(ag)))
+                self.add_object(obj)'''
+
+
+    def get_all_objects(self) -> List['unified_planning.model.object.Object']:
+        '''Returns all the objects.'''
+        return [o for o in self._objects]
+
+
+
+    def add_environment_(self, Env):
+        self.env_ = Env
+
+    def get_environment_(self):
+        return self.env_
+
+    def add_agent(self, Agents):
+        self.agents_list.append(Agents)
+
+    def get_agents(self):
+        return self.agents_list
+
+
+    def compile(self):
+        for flu in self.get_environment_().get_fluents():
+            flu = copy.deepcopy(flu)
+            setattr(flu, '_name', str(getattr(flu, '_name')) + "_env")
```

Comment:
Why? It's 100 better to simply create a new Fluent with a different name (BTW the reason why we do not have a `set_name` is exactly to avoid the need for deep-copies)